### PR TITLE
add enableReturnKeyToSend option

### PR DESCRIPTION
**Pull request by jk2K**
add enableReturnKeyToSend option to using return key to send message

### CODE REVIEW

File: ChattoAdditions/Source/Input/ChatInputBar.swift [hunk 1]
```diff
@@ -250,6 +260,11 @@ extension ChatInputBar: UITextViewDelegate {
     }
 
     public func textView(_ textView: UITextView, shouldChangeTextIn nsRange: NSRange, replacementText text: String) -> Bool {
+        if self.enableReturnKeyToSend && text.range(of: "\n") != nil {
```
Comment (diegosanchezr) [inline ChattoAdditions/Source/Input/ChatInputBar.swift]: This is not working well if you paste text that contains a new line

Comment (dive) [inline ChattoAdditions/Source/Input/ChatInputBar.swift]: More precisely, the text insertion with paragraphs will not work at all :)
I recently solved a similar problem, the code below should work as expected:

```swift
func textView(_ textView: UITextView, shouldChangeTextIn range: NSRange, replacementText text: String) -> Bool {
	let pasteboardString = UIPasteboard.general.string
	let pasteboardStringIsInserted = (pasteboardString == text)
	if (!pasteboardStringIsInserted && (text.rangeOfCharacter(from: .newlines, options: [.backwards])) != nil) {
		// implement desired behavior here
		return false
	}
	return true
}
```

@jk2K 
Perhaps you should add additional checks for array of `strings` stored in `UIPasteboard` or implement `contains(pasteboardTypes:inItemSet:)` check, but the general idea should be clear.

Comment (jk2K) [inline ChattoAdditions/Source/Input/ChatInputBar.swift]: pasting text that contains a new line is allowed.

@dive thanks, perhaps our needs are not the same

**Comment by boboxiaodd** [comment]
it work! paste text also correct
````swift
    class open func loadNib(isReturn:Bool = true) -> ChatInputBar {
        let view = Bundle(for: self).loadNibNamed(self.nibName(), owner: nil, options: nil)!.first as! ChatInputBar
        view.translatesAutoresizingMaskIntoConstraints = false
        view.frame = CGRect.zero
        view.textView.returnKeyType = isReturn ? .default : .send
        view.sendButton.isHidden = !isReturn
        return view
    }

    public func textView(_ textView: UITextView, shouldChangeTextIn nsRange: NSRange, replacementText text: String) -> Bool {
        if text == "\n" && self.sendButton.isHidden {
            self.presenter?.onSendButtonPressed()
            return false
        }
        let range = self.textView.text.bma_rangeFromNSRange(nsRange)
        if let maxCharactersCount = self.maxCharactersCount {
            let currentCount = textView.text.count
            let rangeLength = textView.text[range].count
            let nextCount = currentCount - rangeLength + text.count
            return UInt(nextCount) <= maxCharactersCount
        }
        return true
    }
`````

**Comment by wiruzx** [comment]
Not sure if it's relevant anymore, so closing it for now. Please feel free to reopen

**Pull request closed without merging**